transcom/mymove
# B 20256 int prevent prime users from accessing pptas

## [Agility ticket](tbd)

## Summary

A new field 'allow_pptas' is added to client_certs to allow specified users to use PPTAS. New Authorization method has been added so PPTAS may only be acessed if the aforementioned flag is set for the user's certificate. The flag may be modified in sql directly or within the admin page. Also PPTAS has also been separated from prime's routing. Therefore all pptas routes follow Prime's api naming convention of 'pptas/v1'.


### Setup to Run the Code

- Make sure your postman has the latest devlocal certificate, you will need to update it each time you re-run the server
- Verify that postman has the latest api spec. If not, remove it from postman and re add it using the swagger-def for pptas.
### How to test

Swagger
1. Open the admin page, log in
2. Open the client_certs tab
3. Verify that devlocal certificate has 'Allow PPTAS' checked , otherwise, edit the flag to set it to true.
4. In another tab try to acess the swagger page for pptas at: [PPTAS Swagger](http://primelocal:3000/swagger-ui/pptas.html). if the last step worked, the page should show with no errors
5. Under the same certificate as before, set 'Allow PPTAS' to false
6. Attempt to access [PPTAS Swagger](http://primelocal:3000/swagger-ui/pptas.html)
7. A Forbidden error should be returned

Postman
1. Open Postman
2. Don't modify client_certs (yet)
3. Make a call to the test method of listMoves
4. Because the devlocal certificate should still have 'allow_pptas' as false, you should get a Forbidden result
5. Back in your 'client_cert' tab, set the 'allow_pptas' flag to true
6. Give it a few seconds, then try the making a new request. this time there should be some results
7. Under the same certificate as before, set 'Allow PPTAS' to false
8. Attempt to access [PPTAS Swagger](http://primelocal:3000/swagger-ui/pptas.html)
9. A Forbidden error should be returned
